### PR TITLE
Add Yandex Translate API support

### PR DESCRIPTION
**Pull request by wdv4758h**
* https://tech.yandex.com/translate/
* https://tech.yandex.com/dictionary/

### CODE REVIEW

**Comment by wdv4758h** [comment]
Yandex Dictionary API should fit better to our need, but I can't get the API key, it didn't send the SMS verification code to my phone :P

**Comment by iblis17** [comment]
linter failure on travis

File: zdict/dictionaries/yandex.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import json
+
+from zdict.dictionary import DictBase
+from zdict.exceptions import NotFoundError
+from zdict.models import Record
+
+
+# The daily request limit is 1,000,000 characters. The monthly limit is 10,000,000 characters.
+API_KEY = 'trnsl.1.1.20170826T075621Z.6dfcaff242c6caa8.e2b9cf136d451d9d6eb69516ec97b827e8c8229b'
```
Comment (iblis17) [inline zdict/dictionaries/yandex.py]:
```python
API_KEY = (
    'trnsl.1.1.20170826T075621Z.'
    '6dfcaff242c6caa8.e2b9cf136d451d9d6eb69516ec97b827e8c8229b')

File: zdict/dictionaries/yandex.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    * https://tech.yandex.com/translate/
+    * https://tech.yandex.com/dictionary/
+
+    """
+
+    # Change the url below to the API url of the new dictionary.
+    # Need to keep the `{word}` for `_get_url()` usage.
+    # TODO: support different translate direction
+    # TODO: use Dictionary API
+    API = 'https://translate.yandex.net/api/v1.5/tr.json/translate?key={api_key}&text={word}&lang=ru-en'
```
Comment (iblis17) [inline zdict/dictionaries/yandex.py]: same as previous remark

**Comment by wdv4758h** [comment]
yep modifing code, just open the WIP PR, and I will add a test case later.

**Comment by iblis17** [comment]
there are lots of TODO tag in the code.
mark this PR as WIP?

**Comment by wdv4758h** [comment]
and I think we should update some dependencies :P

**Comment by iblis17** [comment]
> and I think we should update some dependencies :P
maybe in separated  PR, for traceable commit history

File: zdict/dictionaries/yandex.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import json
+
+from zdict.dictionary import DictBase
+from zdict.exceptions import NotFoundError
+from zdict.models import Record
+
+
+# The daily request limit is 1,000,000 characters
+# The monthly limit is 10,000,000 characters
+API_KEY = 'trnsl.1.1.20170826T075621Z.' \
```
Comment (iblis17) [inline zdict/dictionaries/yandex.py]: :/ I'm not a fan of escaping newline.
But I'm ok, if you do not like string concatenation.

**Comment by wdv4758h** [comment]
> > and I think we should update some dependencies :P 
>
> maybe in separated PR, for traceable commit history

yes, I expect it in another PR

File: zdict/dictionaries/yandex.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+          '6dfcaff242c6caa8.e2b9cf136d451d9d6eb69516ec97b827e8c8229b'
+
+
+# Change `Template` to the name of new dictionary. like xxxDict.
```
Comment (iblis17) [inline zdict/dictionaries/yandex.py]: useless comment?

File: zdict/dictionaries/yandex.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        for index, data in enumerate(content['text']):
+            self.color.print('{}. {}'.format(index + 1, data), 'org')
+
+        print()
+
+    def query(self, word: str):
+        content = self._get_raw(word)
+        content_json = json.loads(content)
+
+        if not content_json['code'] != '200':
```
Comment (iblis17) [inline zdict/dictionaries/yandex.py]: why it's combining `not` and `!=` ?

Comment (wdv4758h) [reply zdict/dictionaries/yandex.py]: because it's a copy paste mistake 🍸

**Comment by iblis17** [comment]
what's wrong with Travis on python 3.6? :/

**Comment by iblis17** [comment]
- [ ] test cases

**Comment by wdv4758h** [comment]
Python 3.6 on Travis CI is a shit !?

**Comment by wdv4758h** [comment]
I wonder why I can not get the verification code for the signup of Yandex Dictionary API 😢

**Comment by wdv4758h** [comment]
dude, why Travis CI is angry with Python 3.6 !?

**Comment by iblis17** [comment]
Travis refused to update `setuptools`?

File: zdict/dictionaries/yandex.py [hunk 6]
```diff
@@ -57,8 +67,14 @@ def query(self, word: str):
...
-        if content_json.get('code') != 200:
-            # TODO: handle response codes for different situation
+        status = content_json.get('code')
+        if status != 200:
+            # https://tech.yandex.com/translate/doc/dg/reference/translate-docpage/#codes
+            message = self.status_code.get(
+                status,
+                'Some bad thing happened with Yandex'
+            )
+            print('Yandex: ' + message)
```
Comment (iblis17) [inline zdict/dictionaries/yandex.py]: ` print('Yandex:', message)`?
print will append a space for you.

**Comment by wdv4758h** [comment]
TODO: 

* let user can choose the translate direction, e.g. 'ru-en', 'en-ru', 'es-en', 'en-es'
* support Yandex Dictionary API

these two things won't be done very quick, I'm going to leave it in the future 😝

File: zdict/dictionaries/yandex.py (hunk 6 above)
Comment (wdv4758h) [reply zdict/dictionaries/yandex.py]: I like the concat by `+`, but if you like that one I can change it.

**Comment by wdv4758h** [comment]
zdict-yandex

**Comment by iblis17** [comment]
nice screenshot. add it to README?

**Comment by wdv4758h** [comment]
supported languages in Yandex Translate API: `https://translate.yandex.net/api/v1.5/tr.json/getLangs?key={api_key}`

```
{"dirs":["az-ru","be-bg","be-cs","be-de","be-en","be-es","be-fr","be-it","be-pl","be-ro","be-ru","be-sr","be-tr","bg-be","bg-ru","bg-uk","ca-en","ca-ru","cs-be","cs-en","cs-ru","cs-uk","da-en","da-ru","de-be","de-en","de-es","de-fr","de-it","de-ru","de-tr","de-uk","el-en","el-ru","en-be","en-ca","en-cs","en-da","en-de","en-el","en-es","en-et","en-fi","en-fr","en-hu","en-it","en-lt","en-lv","en-mk","en-nl","en-no","en-pt","en-ru","en-sk","en-sl","en-sq","en-sv","en-tr","en-uk","es-be","es-de","es-en","es-ru","es-uk","et-en","et-ru","fi-en","fi-ru","fr-be","fr-de","fr-en","fr-ru","fr-uk","hr-ru","hu-en","hu-ru","hy-ru","it-be","it-de","it-en","it-ru","it-uk","lt-en","lt-ru","lv-en","lv-ru","mk-en","mk-ru","nl-en","nl-ru","no-en","no-ru","pl-be","pl-ru","pl-uk","pt-en","pt-ru","ro-be","ro-ru","ro-uk","ru-az","ru-be","ru-bg","ru-ca","ru-cs","ru-da","ru-de","ru-el","ru-en","ru-es","ru-et","ru-fi","ru-fr","ru-hr","ru-hu","ru-hy","ru-it","ru-lt","ru-lv","ru-mk","ru-nl","ru-no","ru-pl","ru-pt","ru-ro","ru-sk","ru-sl","ru-sq","ru-sr","ru-sv","ru-tr","ru-uk","sk-en","sk-ru","sl-en","sl-ru","sq-en","sq-ru","sr-be","sr-ru","sr-uk","sv-en","sv-ru","tr-be","tr-de","tr-en","tr-ru","tr-uk","uk-bg","uk-cs","uk-de","uk-en","uk-es","uk-fr","uk-it","uk-pl","uk-ro","uk-ru","uk-sr","uk-tr"]}
```

Currently, I only implement the `ru-en` support.

**Comment by wdv4758h** [comment]
I'll rebase the PR after this one https://github.com/zdict/zdict/pull/128, the testing will be good after that

**Comment by wdv4758h** [comment]
Rebased, let's see the new test result 😝

**Comment by wdv4758h** [comment]
Test passed, but I will leave it here for a few days to let everyone aware this.

**Comment by iblis17** [comment]
oh, you need to update the usage section of README as well.

**Comment by wdv4758h** [comment]
thanks for remind

**Comment by M157q** [comment]
I think we should bump to `0.10.0` after merge this PR?

**Comment by iblis17** [comment]
> I think we should bump to 0.10.0 after merge this PR?

I'm fine with that

**Pull request merged**